test/integration/segmentation_camera_plugin.cc

```diff
@@ -0,0 +1,305 @@
+
+  uint32_t backgroundLabel = 23;
+  camera->SetSegmentationType(SegmentationType::Semantic);
+  camera->EnableColoredMap(false);
+  camera->SetBackgroundLabel(backgroundLabel);
+  camera->SetLocalPosition(0.0, 0.0, 0.0);
+  camera->SetLocalRotation(0.0, 0.0, 0.0);
+
+  // Get the Msg
+  std::string topic =
+    "/test/integration/SegmentationCameraPlugin_imagesWithBuiltinSDF";
+  WaitForMessageTestHelper<ignition::msgs::Image> helper(topic);
+
+  // Update once to create image
+  mgr.RunOnce(std::chrono::steady_clock::duration::zero());
+
+  EXPECT_TRUE(helper.WaitForMessage()) << helper;
+
+  // subscribe to the segmentation camera topic
+  ignition::transport::Node node;
+  node.Subscribe(topic, &OnNewSegmentationFrame);
+
+  // wait for a few segmentation camera frames
+  mgr.RunOnce(std::chrono::steady_clock::duration::zero(), true);
+
+  // wait for a new frame
+  WaitForNewFrame();
+
+  // get the center of each box, the percentages locates the center
+  math::Vector2d leftProj(width * 0.25, height * 0.5);
+  math::Vector2d rightProj(width * 0.75, height * 0.5);
+  math::Vector2d middleProj(width * 0.5, height * 0.5);
+
+  // get their index in the buffer
+  uint32_t leftIndex = (leftProj.Y() * width + leftProj.X()) * 3;
+  uint32_t rightIndex = (rightProj.Y() * width + rightProj.X()) * 3;
+  uint32_t middleIndex = (middleProj.Y() * width + middleProj.X()) * 3;
+
+  // test
+  g_mutex.lock();
+  g_counter = 0;
+
+  // cast the unsigned char to unsigned int to read it
+  unsigned int leftLabel =   g_buffer[leftIndex];
+  unsigned int rightLabel =  g_buffer[rightIndex];
+  unsigned int middleLabel = g_buffer[middleIndex];
+
+  // check the label
+  EXPECT_EQ(leftLabel  , 1);
+  EXPECT_EQ(middleLabel, 2);
+  EXPECT_EQ(rightLabel , 1);
+
+  // check if the first pixel(background) = the background label
+  unsigned int background = g_buffer[0];
+  EXPECT_EQ(background, backgroundLabel);
+
+  g_mutex.unlock();
+
+
+  // Instance/Panoptic Segmentation Test
+  camera->SetSegmentationType(SegmentationType::Panoptic);
+
+  // wait for a few segmentation camera frames
+  mgr.RunOnce(std::chrono::steady_clock::duration::zero(), true);
+  // wait for a new frame
+  WaitForNewFrame();
+
+  // the label in the last channel
+  leftLabel =   g_buffer[leftIndex   + 2];
+  rightLabel =  g_buffer[rightIndex  + 2];
+  middleLabel = g_buffer[middleIndex + 2];
+
+  // the instances count in the first channel
+  unsigned int leftCount =   g_buffer[leftIndex];
+  unsigned int rightCount =  g_buffer[rightIndex];
+  unsigned int middleCount = g_buffer[middleIndex];
+
+  // test
+  g_mutex.lock();
+  g_counter = 0;
+
+  // check the label
+  EXPECT_EQ(leftLabel  , 1);
+  EXPECT_EQ(middleLabel, 2);
+  EXPECT_EQ(rightLabel , 1);
+
+  // instance count
+  EXPECT_EQ(middleCount, 1);
+  EXPECT_EQ(rightCount, 1);
+  EXPECT_EQ(leftCount, 2);
+
+  g_mutex.unlock();
+
+  // Clean up
+  // engine->DestroyScene(scene);
+  // ignition::rendering::unloadEngine(engine->Name());
```

Comment:
remove?

Reply:
Yeah it was a problem that makes the code crashes because of the rendering::SegmentationCamera destructor
It is solved now :D